composer.json

```diff
@@ -6,14 +6,19 @@
     "config": {
         "sort-packages": true,
         "platform": {
-            "php": "8.0.99"
+            "php": "8.1.99"
         },
         "allow-plugins": {
             "dealerdirect/phpcodesniffer-composer-installer": true
         }
     },
+    "conflict": {
+        "amphp/dns": "<2.4.0",
+        "phpunit/phpunit": "<9.6.22",
+        "amphp/socket": "<2.3.1"
+    },
```

Comment:
after some thinking, `conflict` trick for dev-dependencies seems only make sense on "project" type, while on "library", it probably cause issue just to make dev- dependency happy, while user may use different require lib, I think move these to require-dev is better for this

Reply:
Just move the content to require-dev, with change to `^`, I will merge this as is today, and create separate PR to cover this :), ping me if i forgot